### PR TITLE
Build improvements 2023-05-11

### LINKED ISSUE #29: Add caching of GCS data in github actions
Could speed up CI potentially by caching data from GCS. See [here](https://github.com/malariagen/malariagen-data-python/blob/master/.github/workflows/tests.yml) for an example.

### LINKED ISSUE #30: Slow build caused by pathological xarray concatenation
I'm finding the build is slow, especially when trying to work with joined arm contigs. I've pinned it down to some pathological behaviour within xarray's concat() function I think. I don't fully understand what's happening there, but have hacked a workaround in https://github.com/malariagen/malariagen-data-python/pull/395.

### PR DESCRIPTION
**Pull request by alimanfoo**
This PR is aiming to have the build running smoothly and quickly on GCS via datalab-bespin, where we are close to the data and can safely run tasks in parallel.

* [x] Introduce use_gcs_cache config param, to control whether or not to use a local cache of GCS data. 
* [x] Introduce dask_scheduler config param, to control whether dask uses multiple threads.
* [x] Upgrade environment to fix #30.
* [x] Fix #29.

### CODE REVIEW

**Comment by sanjaynagi** [comment]
Thanks @alimanfoo . 

Not sure why the kernel is dying - I just remembered that the CI uses the config.yaml found in .test/ and so is not using joined arms yet (i.e it shouldn't be a memory error).

File: .github/workflows/test-build.yml
Comment (sanjaynagi) [inline .github/workflows/test-build.yml:1]: pretty awesome that you can load a cache in github actions!

**Review by sanjaynagi — APPROVED** [review]
Awesome, thanks @alimanfoo !

**Comment by alimanfoo** [comment]
CI run down to 4 minutes :)

**Comment by alimanfoo** [comment]
Sent you some videos @sanjaynagi via slack.

**Comment by sanjaynagi** [comment]
Awesome, going camping right now so will have to check on Monday! Thank you!!

**Pull request merged**